src/odemis/driver/andorcam2.py

```diff
@@ -2176,118 +2176,132 @@ def _acquire(self):
-                        if self._acq_wait_data(twait):
-                            logging.debug("Acquisition cancelled")
-                            break
-
-                        # Get the data
-                        # In case several images have already been received, we discard all but the last one.
-                        self.atcore.GetMostRecentImage16(cbuffer, c_uint32(im_res[0] * im_res[1]))
-
-                    except (TimeoutError, AndorV2Error) as ex:
-                        # try again up to 5 times
-                        failures += 1
-                        if failures >= 5:
+                        if self.GetStatus() == AndorV2DLL.DRV_ACQUIRING:
+                            self.atcore.AbortAcquisition()
+                            if has_hw_lock:
+                                self.hw_lock.release()
+                                has_hw_lock = False
+                            time.sleep(0.1)  # give a bit of time to abort acquisition
+                    except AndorV2Error as ex:
+                        # it was already aborted
+                        if ex.errno != 20073:  # DRV_IDLE == already aborted == not a big deal
                             raise
-                        # Common failures are 20024 (DRV_NO_NEW_DATA) or
-                        # 20067 (DRV_P2INVALID) during GetMostRecentImage16()
-                        try:
-                            if self.GetStatus() == AndorV2DLL.DRV_ACQUIRING:
-                                self.atcore.AbortAcquisition()  # Need to stop acquisition to read temperature
-                            temp = self.GetTemperature()  # Best way to check the connection and status
-                        except AndorV2Error:  # Probably something really wrong the connection
-                            temp = None
-                        # -999°C means the camera is gone
-                        if temp == -999:
-                            logging.error("Camera seems to have disappeared, will try to reinitialise it")
-                            self.Reinitialize()
-                        else:
-                            time.sleep(0.1)
-                            logging.warning("trying again to acquire image after error %s", ex)
-                        need_reconfig = True
-                        continue  # Go back to beginning of acquisition loop
-                    else:
-                        failures = 0
 
-                    logging.debug("image acquired successfully after %g s", time.time() - tstart)
-                    self.data.notify(self._transposeDAToUser(array))
-                    del cbuffer, array
+                    im_res, duration, trigger_mode = self._update_settings()
+                    if not has_hw_lock:
+                        self.hw_lock.acquire()
+                        has_hw_lock = True
+                    if trigger_mode != TRIG_FAKE:  # TRIG_FAKE uses StartAcquisition to simulate trigger
```

Comment:
Can you remind me why you call startAcquisition when the trigger mode is not fake trigger? The comment suggests the opposite and below in the related elif statement in line 2245 you call the same but when trigger mode equals fake trigger.
Why was it not enough to only start the acquisition (after reconfiguration) in the next block?

Reply:
Normally, we call StartAcquisition() just after reconfiguration, only. As it's in continuous mode, the images will keep coming.
With TRIG_FAKE, we set it to single frame, and use StartAcquisition() to acquire a single image, at the moment the trigger event is received.

I've updated the comment a little to clarify.